### PR TITLE
Changed tree-growing functions in cWorld to use Vector3i coords.

### PR DESCRIPTION
**Pull request by madmaxoft**
The old-style functions are kept in Lua API for backward compatibility, as DeprecatedBindings.

### CODE REVIEW

File: src/World.h [hunk 1]
```diff
@@ -829,30 +829,25 @@ class cWorld  // tolua_export
...
+	Returns true if the tree was grown, false if not (invalid chunk, insufficient space).
+	Exported in DeprecatedBindings due to the obsolete int-based overload and obsolete additional SaplingMeta param. */
+	bool GrowTreeFromSapling(Vector3i a_BlockPos);
 
 	/** Grows a tree at the specified coords, based on the biome in the place.
-	Returns true if the tree was grown, false if not (invalid chunk, insufficient space). */
-	bool GrowTreeByBiome(int a_BlockX, int a_BlockY, int a_BlockZ);
+	Returns true if the tree was grown, false if not (invalid chunk, insufficient space).
+	Exported in DeprecatedBindings due to the obsolete int-based overload. */
+	bool GrowTreeByBiome(const Vector3i a_BlockPos);
```
Comment (tigerw) [inline src/World.h:848]: Could I trouble you to remove the const in the header, also for GrowTree, other files only tag arguments const in cpp files.

**Comment by tigerw** [comment]
Looks good to me! Was that also a bugfix with the `a_Meta & meta` to `meta & 0x07`?

File: src/World.h (hunk 1 above)
Comment (madmaxoft) [reply src/World.h]: Alright.

**Comment by madmaxoft** [comment]
Yes, the meta change is a bugfix. I believe this is what the original author wanted, but failed to implement - the sapling type inside the meta should be the same.

**Comment by madmaxoft** [comment]
This was supposed to be squash-merged :P

**Comment by tigerw** [comment]
Sorry, didn't notice the "Remove const"

But on the other hard, it's never a bad time
```
(•_•)
( •_•)>⌐■-■
(⌐■_■)
````
to see more _flexibility_

**Pull request merged**